### PR TITLE
Use different session IDs for async cache requests

### PR DESCRIPTION
**Pull request by apc999**
- Differentiate `ASYNC_CACHE_WORKER_SESSION_ID` with `ASYNC_CACHE_UFS_SESSION_ID`: the former session is caching from a remote worker and the latter session is caching from UFS
- The goal of this PR is to enable more advanced allocator which can leverage the information: caching from UFS or caching for remote worker to make subtle decisions.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/Sessions.java [hunk 1]
```diff
@@ -33,12 +33,10 @@
 public final class Sessions {
   private static final Logger LOG = LoggerFactory.getLogger(Sessions.class);
 
-  public static final int DATASERVER_SESSION_ID = -1;
-  public static final int CHECKPOINT_SESSION_ID = -2;
   public static final int MIGRATE_DATA_SESSION_ID = -3;
   public static final int MASTER_COMMAND_SESSION_ID = -4;
-  public static final int ACCESS_BLOCK_SESSION_ID = -5;
-  public static final int ASYNC_CACHE_SESSION_ID = -7;
+  public static final int ASYNC_CACHE_WORKER_SESSION_ID = -7;
```
Comment (gpang) [inline core/server/worker/src/main/java/alluxio/Sessions.java]: are these changes backwards compatible?

Comment (apc999) [reply core/server/worker/src/main/java/alluxio/Sessions.java]: Yes, this is backwards compatible

**Comment by apc999** [comment]
@gpang I split this PR into multiple smaller PRs. PTAL

File: core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java [hunk 2]
```diff
@@ -91,8 +91,12 @@ public void submitRequest(AsyncCacheRequest request) {
       mAsyncCacheExecutor.submit(() -> {
         boolean result = false;
         try {
+          boolean isSourceLocal = mLocalWorkerHostname.equals(request.getSourceHost());
```
Comment (gpang) [inline core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java]: How does this distinguish if the block is coming from an alluxio worker, or the ufs?

It looks like it is checking the source of the request of with the local worker hostname, and I'm not sure how if it is UFS or not can be determined by that?

Comment (apc999) [reply core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java]: discussed offline

Comment (gpang) [reply core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java]: O, seems like this code already existed before, and was just moved up to this location.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**